frontend/src/pages/ProgressionChecker/TableView/TableView.jsx

```diff
@@ -1,53 +1,148 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Table } from "antd";
+import { useSelector } from "react-redux";
+import { getMostRecentPastTerm } from "../../TermPlanner/validateTermPlanner";
 import "./index.less";
 
-const TableView = ({ checkercourses }) => {
+const TableView = ({ isLoading, structure }) => {
+  const [pastCourses, setPastCourses] = useState({});
+  const [tableLayout, setTableLayout] = useState({});
+  const { years, startYear, courses } = useSelector((store) => store.planner);
+
+  const getPastCourses = () => {
+    const { Y: recentYear, T: recentTerm } = getMostRecentPastTerm(startYear);
+    const newPastCourses = {};
+
+    years.slice(0, recentYear).forEach((year, yearIndex) => {
```

Comment:
ngl this is a bit confusing haha. Perhaps add a comment explaining some of the logic (no need to be indepth - just a line/few words of comment is good)?

Reply:
ahah true yep will add one